cloudpickle/cloudpickle.py

```diff
@@ -1090,7 +1098,16 @@ def _make_skel_func(code, cell_count, base_globals=None):
     if base_globals is None:
         base_globals = {}
     elif isinstance(base_globals, str):
-        base_globals = vars(sys.modules[base_globals])
+        if base_globals in sys.modules.keys():
+            # this checks if we can import the previous environment the object
+            # lived in
+            base_globals = vars(sys.modules[base_globals])
+        elif id(base_globals) in _BASE_GLOBALS_CACHE.keys():
```

Comment:
You should not use the id of an object that can be garbage collected as a key in a dict: this object id can be reused by Python for a subsequent object.